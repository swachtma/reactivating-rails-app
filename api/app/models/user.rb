# == Schema Information
#
# Table name: users
#
#  id           :integer          not null, primary key
#  github_id    :integer
#  github_email :string
#  username     :string
#  avatar       :string
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#

class User < ApplicationRecord
  has_one :bookmark
  
 def fsa(token, expires)
    {
      id: id,
      github_email: github_email,
      username: username,
      avatar: avatar,
      token: token,
      expires: expires
    }
  end
  
  def self.create_or_fetch(user)
    find_by(github_id: user.id) || create(
      github_id: user.id,
      github_email: user.email,
      username: user.login,
      avatar: user.avatar_url
    )
  end
end