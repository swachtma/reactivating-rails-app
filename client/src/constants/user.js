export const SET_USER = 'SET_USER'
export const SIGNAL_SIGNOUT = 'SIGNAL_SIGNOUT'
export const CLEAR_USER = 'CLEAR_USER'
