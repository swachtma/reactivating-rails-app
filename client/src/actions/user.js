import * as constants from "../constants/user";

export const setUser = (payload) => ({
  type: constants.SET_USER, payload
});

export const clearUser = () => ({ type: constants.CLEAR_USER }) ;