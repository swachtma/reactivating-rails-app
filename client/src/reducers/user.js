import * as constants from '../constants/user';
const default_state = {};

const userReducer = function(state = default_state,action){
  switch (action.type) {
    case(constants.SET_USER):
      return {...action.payload};
    case(constants.CLEAR_USER):
      return {};
    default:
      return state;
  }
};

export default userReducer;