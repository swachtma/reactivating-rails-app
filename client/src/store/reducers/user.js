import * as constants from '../../constants/user';
const default_state = {};

export default function userReducer(state = default_state,action){
  switch (action.type) {
    case(constants.SET_USER):
      return {...action.payload};
    case(constants.CLEAR_USER):
      return {};
    default:
      return state;
  }
}