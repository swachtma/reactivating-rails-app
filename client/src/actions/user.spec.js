/* global expect */
import * as actions from './user'

let test_user = {
  avatar: 'https://avatars0.githubusercontent.com/u/shiba',
  github_email: 'awesome@wow.com',
  id: 9000,
  token: 'wow.much.secret',
  username: 'doge'
}

describe('User actionCreators', () => {
  test('setUser() assigns active user', () => {
    expect(actions.setUser(test_user)).toMatchSnapshot()
  })

  test('clearUser wipes user', () => {
    expect(actions.clearUser()).toMatchSnapshot()
  })
})
