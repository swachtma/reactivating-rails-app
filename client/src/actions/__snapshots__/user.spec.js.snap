// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User actionCreators clearUser wipes user 1`] = `
Object {
  "type": "CLEAR_USER",
}
`;

exports[`User actionCreators setUser() assigns active user 1`] = `
Object {
  "payload": Object {
    "avatar": "https://avatars0.githubusercontent.com/u/shiba",
    "github_email": "awesome@wow.com",
    "id": 9000,
    "token": "wow.much.secret",
    "username": "doge",
  },
  "type": "SET_USER",
}
`;
